src/sentry/models/project.py

```diff
@@ -407,3 +407,40 @@ def get_security_token(self):
 
     def get_lock_key(self):
         return 'project_token:%s' % self.id
+
+    def copy_settings_from(self, project):
+        """
+        Copies project level settings of the inputted project
+        - General Settings
+        - ProjectTeams
+        - Alerts Settings and Rules
+        - EnvironmentProjects
+        - ProjectOwnership Rules and settings
+        - Project Inbound Data Filters
+
+        Returns True if the settings have successfully been copied over
+        Returns False otherwise
+        """
+        from sentry.models import (
+            EnvironmentProject, ProjectOption, ProjectOwnership, Rule
+        )
+        model_list = [EnvironmentProject, ProjectOption, ProjectOwnership, ProjectTeam, Rule]
+
+        try:
+            with transaction.atomic():
```

Comment:
One concern I have is that this transaction might take a while. We should test this out ourselves and see how long it takes to copy Sentry settings to a new project. Then we can make sure that the UI will be responsive and also that we don't hold a transaction for ages and cause issues in the database.

Fine with doing this in production or staging, before we add the UI in.

Reply:
I agree. I honestly think that will be expected from the user... but there's probably front-end magic that can be done. Because there's no frontend code written though adding this to the UI, I'm pretty ok with merging to master. What do you think?